pandas/tests/arrays/boolean/test_construction.py

```diff
@@ -323,17 +323,18 @@ def test_to_numpy_copy():
     tm.assert_extension_array_equal(arr, pd.array([True, False, True], dtype="boolean"))
 
 
-# FIXME: don't leave commented out
 # TODO when BooleanArray coerces to object dtype numpy array, need to do conversion
 # manually in the indexing code
-# def test_indexing_boolean_mask():
-#     arr = pd.array([1, 2, 3, 4], dtype="Int64")
-#     mask = pd.array([True, False, True, False], dtype="boolean")
-#     result = arr[mask]
-#     expected = pd.array([1, 3], dtype="Int64")
-#     tm.assert_extension_array_equal(result, expected)
-
-#     # missing values -> error
-#     mask = pd.array([True, False, True, None], dtype="boolean")
-#     with pytest.raises(IndexError):
-#         result = arr[mask]
+# TODO: belongs somewhere else?
+@pytest.mark.xfail(reason="fails to raise")
+def test_indexing_boolean_mask():
```

Comment:
I think this can simply be removed. I checked git blame, and this was added as commented out test in the initial PR adding the BooleanArray (https://github.com/pandas-dev/pandas/pull/29555), but which didn't yet specify the indexing behaviour. This was only done in a follow-up (https://github.com/pandas-dev/pandas/pull/30308/, https://github.com/pandas-dev/pandas/pull/31591), which added plenty of tests.

(the test is now also outdated)

Reply:
great, will remove